uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/queue/UniProtKBMessageListener.java

```diff
@@ -89,25 +93,47 @@ protected void processMessage(Message message, DownloadJob downloadJob) {
         } else {
             updateDownloadJob(message, downloadJob, JobStatus.RUNNING);
             if (UniProtMediaType.HDF5_MEDIA_TYPE.equals(contentType)) {
-                processH5Message(request, idsFile, jobId);
-                updateDownloadJob(message, downloadJob, JobStatus.UNFINISHED);
+                processH5Message(message, request, downloadJob, idsFile, jobId);
             } else {
                 writeResult(request, idsFile, jobId, contentType);
                 updateDownloadJob(message, downloadJob, JobStatus.FINISHED, jobId);
             }
         }
     }
 
-    private void processH5Message(UniProtKBDownloadRequest request, Path idsFile, String jobId) {
+    private void processH5Message(Message message,
+            UniProtKBDownloadRequest request, DownloadJob downloadJob, Path idsFile, String jobId) {
         try {
-            writeSolrResult(request, idsFile, jobId);
-            sendMessageToEmbeddingsQueue(jobId);
+            Long totalHits = getSolrHits(request);
+            Long maxAllowedHits = this.embeddingsQueueConfigProps.getMaxEntryCount();
+            if (maxAllowedHits >= totalHits) {
+                writeSolrResult(request, idsFile, jobId);
+                sendMessageToEmbeddingsQueue(jobId);
+                updateDownloadJob(message, downloadJob, JobStatus.UNFINISHED);
+            } else {
+                log.warn("Embeddings limit exceeded {}. Max allowed {}", totalHits, maxAllowedHits);
+                updateDownloadJob(
+                        downloadJob,
+                        JobStatus.ABORTED,
+                        String.format(H5_LIMIT_EXCEED_MSG, maxAllowedHits),
+                        downloadJob.getRetried(),
+                        null);
+            }
         } catch (Exception ex) {
             logMessageAndDeleteFile(ex, jobId);
             throw new MessageListenerException(ex);
         }
     }
 
+    private Long getSolrHits(UniProtKBDownloadRequest request) {
```

Comment:
Also there is the queryProcessor that needs to be applied in the requested query.

Reply:
Thanks for pointing out the isoform one.
Regarding "queryProcessor", since i am calling UniProtEntryService.search, does it not have "queryProcessor" set?